rfc/0000-rfc-process.md

```diff
@@ -0,0 +1,60 @@
+---
+name: rfc_process
+started: 2024-07-04
+pr: pact-foundation/roadmap#0000
+---
+## Summary
+
+The Request for Comments (RFC) process is a way to propose and discuss changes to the Pact specification, ecosystem, and community. The RFC process ensures that changes are well-considered, documented, and communicated to all stakeholders. This RFC proposes the process itself, including how to submit an RFC, how to review and discuss it, and how to accept or reject it.
+
+## Motivation
+
+While the evolution of Pact and its ecosystem is open, the process for proposing and discussing changes is not well-defined. By introducing an RFC process, we can ensure that changes are well-considered, documented, and communicated to all stakeholders. This will help to ensure that changes are made in a transparent and collaborative way, and that the community has a voice in the direction of the Pact ecosystem.
+
+## Guide-level explanation
+
+### Scope of RFCs
+
+The Pact ecosystem is broad and includes, broadly speaking, the following areas:
+
+- The Pact specification itself ([pact-foundation/pact-specification](https://github.com/pact-foundation/pact-specification))
+- The Pact broker ([pact-foundation/pact-broker](pact-foundation/pact_broker))
+- The Pact CLI ([pact-foundation/pact-ruby-cli](https://github.com/pact-foundation/pact-ruby-cli))
+- The Pact FFI ([pact-foundation/pact-reference](https://github.com/pact-foundation/pact-reference/tree/master/rust/pact_ffi)).
+
+This RFC process is intended to cover any change which will impact the broader Pact ecosystem.
+
+While specific changes to the Pact implementation in various languages are generally out of scope[^1], we do wish to consider how changes in the areas above will impact the implementation in various languages. Furthermore, we may wish to introduce or formalize common conventions or guidelines that are applied across the various languages to ensure consistency, interoperability, and familiarity for users.
+
+[^1]: Changes to the Pact implementation in various languages are out of scope because they are managed by the maintainers of the individual language implementations (which may even have their own RFC process).
+
+### The Process
+
+The RFC process is intended to remain flexible while still ensuring that changes are well-considered, documented, and communicated to all stakeholders. The process is as follows:
+
+1. Fork the Pact Foundation Roadmap repository: [`https://github.com/pact-foundation/roadmap`](https://github.com/pact-foundation/roadmap)
+2. Copy the `rfc/0000-template.md` file to `rfc/0000-my-feature.md`. Use a simple, descriptive name for the file, but do not assign an RFC number yet.
```

Comment:
How do we organize rfc markdown files?

* We simply put all in one directory? I imagine the directory will be like this:
  * `rfc/0001-do-something.md`
  * `rfc/0005-do-something-else.md` // because PRs from `2` to `4` are rejected
  * `rfc/0006-do-other-thing.md`
  * ...
  * `rfc/9999-a-lot-of-changes.md`
* Do we create different directories (or even multi-level directories) for related rfc? e.g.
  * `rfc/specification/v1/0007-change-to-spec.md`
  * `rfc/specification/v2/00015-change-to-spec.md`
  * `rfc/specification/v2/00037-change-to-spec-again.md`
  * `rfc/broker/0006-change-to-broker.md`
  * `rfc/broker/00022-change-to-broker-again.md`
  * ...

Reply:
In the PR, the person just creates the file `rfc/0000-my-feature.md` and it is only once it is merged that it is assigned a number.

As you point out, yes there will be gaps in the numbering for any PR that is rejected. The numbering is really there to help associate the RFC with the PR that raised the RFC.

As for having directories, I think that might make things more difficult than needed. However, I think it would be a good idea to maintain an index in a separate file to help with discoverability 👍 (though I think that will be up for a later discussion once we have more PRs).